pyvista/core/filters/poly_data.py

```diff
@@ -251,7 +251,17 @@ def boolean_intersection(self, other_mesh, tolerance=1e-5, progress_bar=False):
         See :ref:`boolean_example` for more examples using this filter.
 
         """
-        return self._boolean('intersection', other_mesh, tolerance, progress_bar=progress_bar)
+        bool_inter = self._boolean('intersection', other_mesh, tolerance, progress_bar=progress_bar)
+
+        # check if a polydata is completely contained within another
+        if bool_inter.n_points == 0:
+            inter, s1, s2 = self.intersection(other_mesh)
+            if inter.n_points == 0 and s1.n_points == 0 and s2.n_points == 0:
+                raise RuntimeError(
+                    'Unable to compute boolean intersection when one PolyData is '
+                    'contained within another and no faces intersect.'
+                )
```

Comment:
I think it is a little too harsh to make an error under these conditions. It would be better to warn the user with a warning. What do you think?
Of course, this check is important :)

Reply:
Yea, a warning would be better.

Reply:
FIxed.